### PR TITLE
[WIP] Add Documentation to MNIST example

### PR DESCRIPTION
**Pull request by JeremyFongSP**
Experimenting with documentation for the MNIST example. 
A few things to note:
- Changed title slightly
- Moved step-by-step guide to the bottom for people who just want to c/p the code
- Moved "without NN-ODE" topology part to the explanation section
Please let me know what you think!

### CODE REVIEW

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 1]
```diff
@@ -1,18 +1,17 @@
-# GPU-based Neural ODE Classifier for MNIST with Minibatching
+### Documentation changes:
 
-The following is an example of training a classifier for MNIST using a neural
-ordinary differential equation on GPUs. Note that a few things are demonstrated
-in this example. One good trick to know is:
+* Short Title with keywords
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:3]: these kinds of comments should go in the PR but not in the changed code

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 2]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+       (iter%10 == 0) && @show accuracy(m, zip(x_train,y_train))
+       global nfe=0
 end
 
 # Train the NN-ODE and monitor the loss and weights.
 Flux.train!( loss, params( down, nn_ode.p, fc), zip( x_train, y_train ), opt, cb = cb )
 ```
+
+
+## Step-by-Step
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:116]:
```suggestion
## Step-by-Step Description
```

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 3]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+
+### Packages
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:118]:
```suggestion
### Load Packages
```

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 4]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+	return x_train, y_train
+end
+```
+
+And then loaded from main
+```
+# Main
+const bs = 128
+x_train, y_train = loadmnist(bs)
+```
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:174]: very nicely written section

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 5]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+fc  = Chain(Dense(20,10)) |> gpu
+
+nfe = 0
+```
+
+`down`: Takes a 28x28 image, flattens it, then passes it through a fully connected
+layer with `tanh` activation<br>
+`nn`: Neural network chain three layers deep, `tanh` activation <br>
+`nn_ode`: ODE solver for neural network <mark><-- Should add more explanation </mark><br>
+`fc`: Final fully connected layer <br>
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:209]: The <br> aren't needed: this is markdown not html

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 6]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+
+fc  = Chain(Dense(20,10)) |> gpu
+
+nfe = 0
+```
+
+`down`: Takes a 28x28 image, flattens it, then passes it through a fully connected
+layer with `tanh` activation<br>
+`nn`: Neural network chain three layers deep, `tanh` activation <br>
+`nn_ode`: ODE solver for neural network <mark><-- Should add more explanation </mark><br>
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:208]: that's probably fine. You can probably just say that this is the ODE solver layer

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 7]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+    xarr = gpu(x)
+    return reshape(xarr, size(xarr)[1:2])
+end
+```
+
+This is required to convert the output of the Neural Network ODE.
+
+### Build Topology
+
+Connect all layers together in a single chain
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:228]: I think it would be good if this has a narrative flow. "Next we connect all of the layers together ..."

File: docs/src/examples/Supervised-NN-ODE-MNIST.md [hunk 8]
```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
...
+loss(x_train[1],y_train[1])
+```
+
+### Optimizer
+```julia
+opt = ADAM(0.05)
+iter = 0
+```
+
+### <mark>Unsure</mark>
```
Comment (ChrisRackauckas) [inline docs/src/examples/Supervised-NN-ODE-MNIST.md:300]: this is a callback for printing the loss during the training. It prints the current accuracy on the training set every 10 iterations by updating `iter` as a global each step and then doing `@show accuracy...`

**Comment by ChrisRackauckas** [comment]
Nice! I think this is a fantastic improvement. I like the "code on top, explanation below" style.

File: docs/src/examples/Supervised-NN-ODE-MNIST.md (hunk 8 above)
Comment (JeremyFongSP) [reply docs/src/examples/Supervised-NN-ODE-MNIST.md]: That makes sense. What is nfe used for here?

Comment (ChrisRackauckas) [reply docs/src/examples/Supervised-NN-ODE-MNIST.md]: number of function evaluations. That's used to count the number of times that the callback is called so it only prints the loss every 10th time.

**Comment by ChrisRackauckas** [comment]
Those are some fantastic improvements. They can keep coming in another PR as well, but this is so much better already that I wanted to merge so it can start being shared.
